BackEnd/Core/sailfish-common/build.gradle

```diff
@@ -50,13 +60,33 @@ dependencies {
 
 task templates {
     doLast {
-	    ant.taskdef(name: 'jamon', classname: 'org.jamon.ant.JamonTask',   classpath: configurations.jamon.asPath)
-	    ant.taskdef(name: 'xjc',   classname: 'com.sun.tools.xjc.XJCTask', classpath: configurations.jaxb.asPath)
-	    ant.jamon(destdir: "${genJavaDir}", srcdir: "${templatesDir}")
-        ant.xjc(schema: "${xsdDir}/dictionary.xsd", destdir: "${genJavaDir}", package: "com.exactpro.sf.common.impl.messages.xml.configuration", removeOldOutput: "yes") {
-            produces(dir: "${genJavaDir}/com/exactpro/common/sf/impl/messages/xml/configuration", includes: "*")
-	        arg(value: "-extension")
-	    }
+        ant.taskdef(name: 'jamon', classname: 'org.jamon.ant.JamonTask',   classpath: configurations.jamon.asPath)
+        ant.taskdef(name: 'xjc',   classname: 'com.sun.tools.xjc.XJCTask', classpath: configurations.jaxb.asPath)
+        ant.jamon(destdir: "${genJavaDir}", srcdir: "${templatesDir}")
+        ant.xjc(schema: "${xsdDir}/dictionary.xsd", destdir: "${genJavaDir}", package: "com.exactpro.sf.common.impl.messages.xml.configuration") {
+            produces(dir: "${genJavaDir}/com/exactpro/sf/common/impl/messages/xml/configuration", includes: "*")
+            arg(value: "-extension")
+        }
+    }
+}
+
+task schemaGen () {
+    doLast {
+        ant.taskdef(name: 'schemagen', classname: 'com.sun.tools.jxc.SchemaGenTask', classpath: configurations.jaxb.asPath)
+        ant.schemagen(
+                srcdir: "${mainJavaDir}",
+                destdir: "${xsdDir}",
+                includeAntRuntime:'false'
+        ) {
+            // jxc needs to know the package name
+            include(name: 'com/exactpro/sf/common/impl/messages/all/configuration/sample/')
+
+            // jxc needs to know other sources such as interfaces in our case
+            classpath {
+                pathElement(path: configurations.sailfish_common.asPath )
+            }
+            schema(file: "generateddictionary.xsd", namespace: "http://exactprosystems.com/dictionary")
+        }
     }
 }
```

Comment:
We've figured out that the task should look like this:
<pre>
task schemaGen () {
    dependsOn.add('compileKotlin')
    doLast {
        ant.taskdef(name: 'schemagen', classname: 'com.sun.tools.jxc.SchemaGenTask', classpath: configurations.jaxb.asPath)
        ant.schemagen(
                srcdir: "${mainJavaDir}",
                destdir: "${xsdDir}",
                includeAntRuntime:'false',
                verbose: 'false'
        ) {
            // jxc needs to know the package name
            include(name: 'com/exactpro/sf/common/impl/messages/all/configuration/sample/')

            // jxc needs to know other sources such as interfaces in our case
            classpath {
                pathElement(path: sourceSets.main.output.classesDirs.asPath )
            }
            schema(file: "generateddictionary.xsd", namespace: "http://exactprosystems.com/dictionary")
        }
    }
}

processResources.dependsOn 'schemaGen'
</pre>

Main point:
1. **dependsOn.add('compileKotlin')** - because we need compiled classes
2. **pathElement(path: sourceSets.main.output.classesDirs.asPath )** - because we need compiled classes
3. **processResources.dependsOn 'schemaGen'** - because we need schema in the resources